pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -657,3 +655,30 @@ func getNodeMAC(node *corev1.Node) (net.HardwareAddr, error) {
 	}
 	return mac, nil
 }
+
+func (c *Controller) getNodeIPForPods(node *corev1.Node) (net.IP, error) {
+	if c.networkConfig.TransportIface != "" {
+		peerNodeIP, err := getNodeTransportAddress(node)
+		if err != nil {
+			klog.Errorf("Failed to retrieve transport IP address of Node %s: %v", node.Name, err)
+			return nil, err
```

Comment:
Can it fall back to use Node IP if the annotation is not found? otherwise upgrade or version skew may be disruptive.

Reply:
You could think about how a cluster is upgraded: yaml is applied and daemonset pods are rolling updated.
After the 1th agent is updated, no other agents have reported transport address, so all traffic to other Nodes would break.
After the 2nd agent is updated, pods on the 1th node can connect to pods on the 2th node but no others.
...
After the last agent updated, pods on the 1th node can connect to pods on it.
Updating an agent may take tens of seconds, so you can imagine how long the disruption would last.

The route will be overwritten when a Node's transportAddress is updated, would there be a problem except traffic would go to management interface temporarily (I think this is a very short period if it's a fresh installation as all agents are installed at the same time). If it's an upgrade or configuration change, I think it's toally fine to use management interface for a while as anyway it was used before the upgrade or configuration change.

Reply:
For Windows, it might introduces a wrong configuration, two routing entry will be added. Besides, a misconfiguration in OpenFlow entries might exist: Reply to Linux management IP is possible to be output from the uplink interface (transport interafce)

Reply:
@jianjuns @antoninbas Could you share your points?

Reply:
I haven't been following this PR closely but I think Quan's concern is valid.
Generally, I don't think we should have a strong dependency on this annotation (upgrade is an example of this) and we should always be able to fallback to the Node IP advertised by K8s. It seems that this fallback will not help on Windows because the transport interface becomes the uplink and may not be able to reach other Nodes using their management IP. However, we will at least avoid disruptions for Linux clusters.

Reply:
I also feel fallback to mgmt IP is better. However, I did not get why upgrade can run into the issues. Even after upgrade, by default we do not set the transportInterface parameter?

Reply:
Got it, then I would change.

Reply:
@jianjuns Yes, upgrading without changing any configuration wouldn't hit the issue. The case in my mind is updating the yaml to specify the interface with Node IP explicitly or switch to another interface.